src/views/PeopleSearch/components/SearchFieldList/index.tsx

```diff
@@ -0,0 +1,493 @@
+import {
+  Accordion,
+  AccordionDetails,
+  AccordionSummary,
+  Button,
+  Card,
+  CardActions,
+  CardContent,
+  CardHeader,
+  Checkbox,
+  FormControlLabel,
+  FormLabel,
+  Grid,
+  Typography,
+} from '@material-ui/core';
+import { ExpandMore } from '@material-ui/icons';
+import GordonLoader from 'components/Loader';
+import { useAuthGroups, useUser } from 'hooks';
+import {
+  ChangeEvent,
+  Dispatch,
+  KeyboardEvent,
+  SetStateAction,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import {
+  FaBook,
+  FaBriefcase,
+  FaBuilding,
+  FaGlobeAmericas,
+  FaHeart,
+  FaSchool,
+  FaHome as Home,
+  FaUser as Person,
+  FaMapMarkerAlt as LocationCity,
+} from 'react-icons/fa';
+import { useHistory, useLocation } from 'react-router';
+import { AuthGroup } from 'services/auth';
+import peopleSearchService, { Class, PeopleSearchQuery, SearchResult } from 'services/peopleSearch';
+import { toTitleCase, searchParamSerializerFactory } from 'services/utils';
+import { gordonColors } from 'theme';
+import SearchField from './components/SearchField';
+
+/**
+ * A Regular Expression that matches any string with any alphanumeric character `[a-z][A-Z][0-9]`.
+ * See [RegExp Character Classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Character_Classes#types) for more info.
+ */
+const containsLetterRegExp = /\w/;
+
+const relationship_statuses = [
+  'Single',
+  'Taken',
+  'Engaged',
+  'Married',
+  'At DTR Bench Right NOW',
+  '1st DTR',
+  '2nd DTR',
+  'Sat Together At Chapel',
+  'Jesus Is My Significant Other',
+  'Waiting For Her Boaz',
+  'Waiting For His Proverbs 31 Woman',
+  "It's Complicated",
+];
+
+const searchPageTitle = (
+  <>
+    Search the
+    <b style={{ color: gordonColors.primary.cyan }}> Gordon </b>
+    Community
+  </>
+);
+
+const defaultSearchParams: PeopleSearchQuery = {
+  includeStudent: true,
+  includeFacStaff: true,
+  includeAlumni: false,
+  first_name: '',
+  last_name: '',
+  major: '',
+  minor: '',
+  residence_hall: '',
+  class_year: '',
+  home_town: '',
+  state: '',
+  country: '',
+  department: '',
+  building: '',
+};
+
+const { serializeSearchParams, deserializeSearchParams } =
+  searchParamSerializerFactory(defaultSearchParams);
+
+const isTodayAprilFools = () => {
+  const todaysDate = new Date();
+  return todaysDate.getMonth() === 3 && todaysDate.getDate() === 1;
+};
```

Comment:
Technically, this introduces a bug where we create a new `Date` twice, and the second one will possibly have a later date than the first. In reality, that shouldn't matter because the expected difference in time between the two dates is at most milliseconds.

My other concern would be that, if we statically calculate this outside the component, it might be evaluated at build time (or some other point), rather than being recalculated on render, meaning the value would become stale and inaccurate.